File: eva-core/src/main/java/io/nerv/core/upload/util/FileUploadProvider.java
package io.nerv.core.upload.util;

import org.springframework.web.multipart.MultipartFile;

import java.io.File;
import java.io.IOException;
import java.util.List;

/**
 * 图片上传接口
 */
public interface FileUploadProvider {
    /**
     * 上传
     * @param image
     * @return
     */
    String upload(MultipartFile image);

    /**
     * 存储
     * @param filenames
     * @return
     */
    List<String> storage(String... filenames);

    /**
     * 存储图片并且生成缩略图
     * @param scale
     * @param filenames
     */
    void storageWithThumbnail(float scale, String... filenames);

    /**
     * 生成缩略图
     * @param file
     * @param scale
     */
    void thumbnail(File file, float scale);

    /**
     * 生成缩略图,可指定缩放
     * @param file
     * @param dest
     * @param scale
     */
    void thumbnail(File file, File dest, float scale);

    /**
     * 删除图片
     * @param fileName
     */
    void delFromStorage(String fileName);
}

File: eva-core/src/main/kotlin/io/nerv/core/upload/util/FileUploadProvider.kt
package io.nerv.core.upload.util

import org.springframework.web.multipart.MultipartFile
import java.io.File

/**
 * 文件上传接口
 */
interface FileUploadProvider {
    /**
     * 上传
     * @param file
     * @return
     */
    fun upload(file: MultipartFile, path: String?): String?

    /**
     * 存储
     * @param filenames
     * @return
     */
    fun storage(vararg filenames: String?): List<String?>?

    /**
     * 存储图片并且生成缩略图
     * @param scale
     * @param filenames
     */
    fun storageWithThumbnail(scale: Float, vararg filenames: String)

    /**
     * 生成缩略图
     * @param file
     * @param scale
     */
    fun thumbnail(file: File, scale: Float)

    /**
     * 生成缩略图,可指定缩放
     * @param file
     * @param dest
     * @param scale
     */
    fun thumbnail(file: File?, dest: File?, scale: Float)

    /**
     * 删除图片
     * @param fileName
     */
    fun delFromStorage(fileName: String)

    /**
     * 缓存清除
     */
    fun tempClean()
}